src/main/java/com/celements/pagetype/service/PageTypeService.java

```diff
@@ -66,13 +66,15 @@
   private List<IPageTypeCategoryRole> pageTypeCategoryList;
 
   private final ConcurrentMap<String, IPageTypeCategoryRole> typeNameToCatCache = new ConcurrentHashMap<>();
+  private final ConcurrentMap<String, IPageTypeCategoryRole> typeNameToCatCacheDeprecated = new ConcurrentHashMap<>();
 
   @Override
   public void resetTypeNameToCatCache() {
     typeNameToCatCache.clear();
+    typeNameToCatCacheDeprecated.clear();
   }
 
-  private Map<String, IPageTypeCategoryRole> getTypeNameToCategoryMap() {
+  private Map<String, IPageTypeCategoryRole> getTypeNameToCategoryMap(boolean deprecated) {
```

Comment:
please do not add a boolean to the getTypeNameToCategoryMap. This binds the implementation to the interface. Instead rename getTypeNameToCategoryMap to void loadTypeNameToCategoryMap. Then implement getTypeNameToCategoryMap returning only the not deprecated names (as before) and getTypeNameToCategoryMapIncludeDeprecated returning typeNameToCatCacheDeprecated. Call the loadTypeNameToCategoryMap at the beginning of both get methods.